File: solutions/java/gigasecond/1/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

public class Gigasecond {
    private LocalDateTime init;

    public Gigasecond(LocalDate moment) {
        init = LocalDateTime.of(moment, LocalTime.MIDNIGHT);
    }

    public Gigasecond(LocalDateTime moment) {
        init = moment;
    }

    public LocalDateTime getDateTime() {
        return init.plusSeconds(1_000_000_000);
    }    
}

File: solutions/kotlin/gigasecond/1/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime

class Gigasecond {
  var date: Any

  constructor(localDateTime: LocalDateTime) {
    date = localDateTime.plusSeconds(1_000_000_000)
  }

  constructor(localDate: LocalDate) {
    val of = LocalDateTime.of(localDate.year, localDate.month, localDate.dayOfMonth, 0, 0)
    date = of.plusSeconds(1_000_000_000)
  }
}
